### PR TITLE
feat/docs/test: Updated linear search and added comments and test cases

### PR DESCRIPTION
**Pull request by shashirajraja**
#### Description of Change
* Added test cases for self-test of the linear search function 
* Added appropriate comments and description
* Updated the linear search to return location of the element if found else -1 

#### References
https://en.wikipedia.org/wiki/Linear_search

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/889"></a>

### CODE REVIEW

File: searching/linear_search.c [hunk 1]
```diff
@@ -1,36 +1,67 @@
-#include <stdio.h>
-#include <stdlib.h>
+/**
+ * @file
+ * @brief Algorithm to Search an Element from the array
```
Comment (Panquesito7) [inline searching/linear_search.c:3]: Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂

File: searching/linear_search.c [hunk 2]
```diff
@@ -1,36 +1,67 @@
...
+	
+	assert(linearsearch(arr, 16, 52) == 3);
+	assert(linearsearch(arr, 16, 10) == 1);
+	assert(linearsearch(arr, 16, 6) == 11);
+	assert(linearsearch(arr, 16, 55) == 16);
+	assert(linearsearch(arr, 16, 19) == 5);
+	assert(linearsearch(arr, 16, 36) == 6);
+	assert(linearsearch(arr, 16, 91) == -1);
+    
+	printf("All test cases passed successfully");
```
Comment (Panquesito7) [inline searching/linear_search.c:56]:
```suggestion
	printf("All test cases have successfully passed!\n");
```

**Review by Panquesito7 — COMMENTED** [review]
Almost there! 😄

File: searching/linear_search.c (hunk 1 above)
Comment (shashirajraja) [reply searching/linear_search.c:3]: Done..!
Hope now it meets the requirements😊

Thanks !!

**Comment by Panquesito7** [comment]
Any reasons why it's failing? @straight-into-the-wall.

**Pull request closed without merging**